### PR TITLE
issue 24: Specifications with `@Testcontainers` annotation can now subclass each other

### PR DESCRIPTION
**Pull request by iNikem**

### CODE REVIEW

**Comment by iNikem** [comment]
Hm, those tests failed locally as well. May because of changing testcontainers version from 1.4.1 to 1.4.2. But I cannot test with 1.4.1 as it fails on my old MacOS. @kiview can you test this PR with 1.4.1?

**Comment by iNikem** [comment]
Sorry for all confusion :)

File: src/main/groovy/com/groovycoder/spockdockerextension/TestcontainersMethodInterceptor.groovy [hunk 1]
```diff
@@ -74,15 +72,15 @@ class TestcontainersMethodInterceptor extends AbstractMethodInterceptor {
     }
 
     private static void startContainers(List<FieldInfo> containers, IMethodInvocation invocation) {
-
         containers.each { FieldInfo f ->
             GenericContainer container = readContainerFromField(f, invocation)
-            container.start()
+            if(!container.isRunning()){
```
Comment (bsideup) [inline src/main/groovy/com/groovycoder/spockdockerextension/TestcontainersMethodInterceptor.groovy:77]: we probably should do this check in `.start()` /cc @rnorth @kiview

Comment (kiview) [reply src/main/groovy/com/groovycoder/spockdockerextension/TestcontainersMethodInterceptor.groovy:77]: Would make this call idempotent. Is this a problem with `.stop()` as well?

**Comment by kiview** [comment]
Fix #24

File: src/main/groovy/com/groovycoder/spockdockerextension/TestcontainersMethodInterceptor.groovy (hunk 1 above)
Comment (iNikem) [reply src/main/groovy/com/groovycoder/spockdockerextension/TestcontainersMethodInterceptor.groovy:77]: Stop method in `GenericContainer` already has a protection against double stopping

**Pull request merged**